src/encrypt.js

```diff
@@ -208,16 +230,13 @@ function hkdf(salt, ikm, info, length) {
 function encryptPayload(plaintext, contentEncryptionKey, nonce) {
   const cipher = crypto.createCipheriv('id-aes128-GCM', contentEncryptionKey,
       nonce);
-  // TODO: This is the minimum padding buffer. We could instead use this to
-  // obscure the length of the payload by choosing a different padding length
-  // based on the length of the plaintext.
-  const paddingResult = cipher.update(PADDING_BUFFER);
-  const textResult = cipher.update(plaintext);
+  const result = cipher.update(plaintext);
   cipher.final();
 
-  return Buffer.concat([paddingResult, textResult, cipher.getAuthTag()]);
+  return Buffer.concat([result, cipher.getAuthTag()]);
 }
 
 // All functions are exported here to make them testable, but only `encrypt` is
 // re-exported by `index.js` as part of the public API.
-module.exports = {encrypt, createContext, createInfo, hkdf, encryptPayload};
+module.exports = {encrypt, createContext, createInfo, hkdf, makePadding,
```

Comment:
I'm slightly cautious of what is being exposed and the use case for it. This is feeling more and more like it could just be a class with static methods.